js/client.js

```diff
@@ -913,6 +913,8 @@ function toId() {
 		 * Receive from sim server
 		 */
 		receive: function (data) {
+			//trigger popup lisener when messege is received from server and save to storeg.pref obj  
+			app.trigger("lisenToPopUpMessege",{msg:data});
```

Comment:
This is not necessarily accurate - this can be things other than a popup. That seems unintentional. I'd wait to trigger this until the `switch` statement later down that checks message types. Just call it in the `popup` case.